testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/actions/AppInitiatedActionWebAuthnTest.java

```diff
@@ -0,0 +1,127 @@
+import org.keycloak.representations.idm.RealmRepresentation;
+import org.keycloak.representations.idm.RequiredActionProviderRepresentation;
+import org.keycloak.testsuite.WebAuthnAssume;
+import org.keycloak.testsuite.arquillian.annotation.EnableFeature;
+import org.keycloak.testsuite.pages.LoginUsernameOnlyPage;
+import org.keycloak.testsuite.pages.PasswordPage;
+import org.keycloak.testsuite.pages.webauthn.WebAuthnRegisterPage;
+import org.keycloak.testsuite.util.FlowUtil;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.keycloak.common.Profile.Feature.WEB_AUTHN;
+import static org.keycloak.models.AuthenticationExecutionModel.Requirement.ALTERNATIVE;
+import static org.keycloak.models.AuthenticationExecutionModel.Requirement.REQUIRED;
+
+/**
+ * @author <a href="mailto:mabartos@redhat.com">Martin Bartos</a>
+ */
+@EnableFeature(value = WEB_AUTHN, skipRestart = true, onlyForProduct = true)
+public class AppInitiatedActionWebAuthnTest extends AbstractAppInitiatedActionTest {
+
+    @Page
+    LoginUsernameOnlyPage usernamePage;
+
+    @Page
+    PasswordPage passwordPage;
+
+    @Page
+    WebAuthnRegisterPage registerPage;
+
+    public AppInitiatedActionWebAuthnTest() {
+        super(WebAuthnRegisterFactory.PROVIDER_ID);
+    }
+
+    @Override
+    protected boolean isImportAfterEachMethod() {
+        return true;
+    }
+
+    @Override
+    public void configureTestRealm(RealmRepresentation testRealm) {
+        RequiredActionProviderRepresentation action = new RequiredActionProviderRepresentation();
+        action.setAlias(WebAuthnRegisterFactory.PROVIDER_ID);
+        action.setProviderId(WebAuthnRegisterFactory.PROVIDER_ID);
+        action.setEnabled(true);
+        action.setDefaultAction(true);
+        action.setPriority(10);
+
+        List<RequiredActionProviderRepresentation> actions = new ArrayList<>();
+        actions.add(action);
+        testRealm.setRequiredActions(actions);
+    }
+
+    @Before
+    public void setUpWebAuthnFlow() {
+        final String newFlowAlias = "browserWebAuthnAIA";
+        testingClient.server("test").run(session -> FlowUtil.inCurrentRealm(session).copyBrowserFlow(newFlowAlias));
+        testingClient.server("test").run(session -> {
+            FlowUtil.inCurrentRealm(session)
+                    .selectFlow(newFlowAlias)
+                    .inForms(forms -> forms
+                            .clear()
+                            .addAuthenticatorExecution(REQUIRED, UsernameFormFactory.PROVIDER_ID)
+                            .addSubFlowExecution(REQUIRED, subFlow -> subFlow
+                                    .addAuthenticatorExecution(ALTERNATIVE, PasswordFormFactory.PROVIDER_ID)
+                                    .addAuthenticatorExecution(ALTERNATIVE, WebAuthnAuthenticatorFactory.PROVIDER_ID)))
+                    .defineAsBrowserFlow();
+        });
+    }
+
+    @Before
+    public void verifyEnvironment() {
+        WebAuthnAssume.assumeChrome();
```

Comment:
AFAIK with this assume, the test will be never executed anywhere. I guess we have 2 options:
- If test currently doesn't require active WebAuthn support and will work by default with HtmlUnit and all the other browsers without WebAuthn support (which seems so), then we can remove the assume for now, so the test is executed in travis, pipeline and all the other places
- Otherwise we can add it to different package etc, so that it is executed in the pipeline during running of WebAuthn tests.

Reply:
I would choose the second option, because the test, lately, should verify functionality of the WebAuthn AIA and the WebAuthn feature will be necessary.

Reply:
Cool, Thanks